src/runtime_src/hip/api/hip_event.cpp

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023-2024 Advanced Micro Device, Inc. All rights reserved.
+
+#include "core/common/error.h"
+
+#include "hip/config.h"
+#include "hip/hip_runtime_api.h"
+
+#include "hip/core/event.h"
+
+namespace xrt::core::hip {
+static hipEvent_t
+hip_event_create()
+{
+  throw std::runtime_error("Not implemented");
+}
+
+static void
+hip_event_destroy(hipEvent_t event)
+{
+  if (!event)
+    throw xrt_core::system_error(hipErrorInvalidHandle, "event is nullptr");
```

Comment:
Refactor the code to avoid repetitions, for example with
```suggestion
  throw_if_null(!event);
```